entity-framework/core/querying/complex-query-operators.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Complex Query Operators - EF Core
+author: smitpatel
+ms.date: 09/30/2019
+ms.assetid: 2e187a2a-4072-4198-9040-aaad68e424fd
+uid: core/querying/complex-query-operators
+---
+# Complex Query Operators
+
+Language Integrated Query (LINQ) contains many complex operators which combines multiple data sources or does complex processing. Not all LINQ operators have suitable translation in server side. Sometimes, query written in one form can be translated but in different form doing different computation cannot be translated. This page describes some of the complex operators and their supported variations. In future we may recognized more pattern and add their translation. Important point to remember is database also varies in what they support so a particular query which is translated in SqlServer may not work for Sqlite database.
+
+> [!TIP]
+> You can view this article's [sample](https://github.com/aspnet/EntityFramework.Docs/tree/master/samples/core/Querying) on GitHub.
+
+## Join
+
+LINQ Join operator allows you to connect two data sources based on the key selector on each source and generates a tuple of values when key matches. This naturally translates to `INNER JOIN` on relational databases. While LINQ Join has outer & inner key selectors, databases take a condition as join condition and EF Core will generate join condition by comparing outer key selector to inner key selector through equality operation. Further, if key selectors are anonymous types, EF Core will generate join condition to perform equality component wise.
+
+[!code-csharp[Main](../../../samples/core/Querying/ComplexQuery/Sample.cs#Join)]
+
+```SQL
+SELECT [p].[PersonId], [p].[Name], [p].[PhotoId], [p0].[PersonPhotoId], [p0].[Caption], [p0].[Photo]
+FROM [PersonPhoto] AS [p0]
+INNER JOIN [Person] AS [p] ON [p0].[PersonPhotoId] = [p].[PhotoId]
+```
+
+## GroupJoin
+
+LINQ GroupJoin operator allows you to connect two data sources in same way as Join but the different here is, it creates a group of values from inner source which matches key with element from outer source. Performing query like following generates result of `Blog` & `IEnumerable<Post>`. Since databases (especially relational databases) do not support representing a collection of client side objects, GroupJoin cannot be translated to server in many cases. It requires to get all the data from server to perform GroupJoin without special selector. But if the selector is limiting data being selected then fetching all data from server may cause performance issues. Hence EF Core does not translate GroupJoin.
```

Comment:
```suggestion
LINQ GroupJoin operator allows you to connect two data sources in same way as Join but the difference here is that it creates a group of values from inner source which matches the key with an element from the outer source. Performing a query like the following generates result of `Blog` & `IEnumerable<Post>`. Since databases (especially relational databases) do not support representing a collection of client side objects, GroupJoin cannot be translated to server in many cases. It requires to get all the data from server to perform GroupJoin without a special selector. But if the selector is limiting data being selected then fetching all data from server may cause performance issues. Hence EF Core does not translate GroupJoin.
```